src/viem/saga.ts

```diff
@@ -99,13 +101,16 @@ export function* sendPayment({
     yield* call(unlockAccount, wallet.account.address)
   }
 
+  const feeCurrency: string | undefined = preparedTransaction && getFeeCurrency(preparedTransaction)
```

Comment:
This way we avoid adding another variant of `getFeeCurrency` for `SerializableTransactionRequest`?
```suggestion
  const feeCurrency: string | undefined = preparedTransaction && getFeeCurrency(getPreparedTransaction(preparedTransaction))
```